### DISCUSSION TITLE
How to activate "Gate step (V) and Run to break (B)"  in Macos version ?

### DISCUSSION THREAD

**Discussion by jongena** [hneemann/Digital#1065] [Q&A]
"Gate step (button V)" seems not to be activated in the Macos version ? 
Same for "Run to break in single gate mode (button B) ?
They are always gray....? 

How to make them "active"  ?

**Comment by hneemann** [comment]
You have to start the simulator in gate step mode (button G).

**Comment by jongena** [comment]
Thank you for the answer. But typing the G button in de Mac version seems not to work. I suppose that this should be equivalent to the 
 button on the top of the window, called "runs the circuit in single gate step mode". That works, but then the two buttons to the right
"Gate step" and "Run to break in single gate mode" remain grayed out, so I cannot click on them. For the OS, Java and Digital version I have here : Macos Version 12.6.1, JRE: 1.8.0_351, VM 25.351-b10, git-Revision v0.29, Build time 2022-02-11 17:13
From what I find on the website this seems to be indeed the latest Digital version. Also Java is the latest available version for Macos.
Anything else I forgot to install / check ?

**Comment by hneemann** [comment]
Hmm! There is nothing else to install. As soon as one of the inputs changes, the affected gates should be highlighted and the buttons should be clickable.

**Comment by xotmatrix** [comment]
I can't figure out how to get this feature to work either in v0.29. I'm on Windows. The documentation has nothing say on it, except the keyboard shortcut says "V" not "G". This matches the menu. When "Single gate stepping" is started, "Gate step" and "Run to Break..." are disabled in the menu and the button bar.

![javaw_2022-12-02_11-51-06](https://user-images.githubusercontent.com/14209329/205343924-e471da88-0784-4450-97fa-9d3f2f2ea85f.png)

**Reply by hneemann** [reply]
@xotmatrix 
Could you try the latest pre release?
That works perfectly for me.

**Reply by xotmatrix** [reply]
It appears to behave exactly the same. I cannot get it to work in an obvious way.

![javaw_2022-12-02_13-01-11](https://user-images.githubusercontent.com/14209329/205356937-eafea3d0-a9fb-4316-85f3-ca70143457c6.png)

**Reply by xotmatrix** [reply]
OK, new information. It does possibly work as intended, I'm not sure. 

I was playing with some earlier releases and it behaved the same. 

This circuit uses a real-time clock. The "Gate Step" and "Run to Break" functions only become enabled when I click on the clock component to advance it (pressing "C" does not work in this mode -- edit: this was caused by there being two clocks, which I know is not the right way. After removing one of the clocks, "C" advances the clock _but does not enable the functions_). Then after it completes a series of steps, the functions become disabled again until another clock transition is generated manually.

Since Digital is event-based, it seems like maybe this is the intended operation. It won't simulate a step if there is no state change from the clock (or some other input or component). Should a real-time clock function while in gate step mode is an open question. 

Stepping functions not becoming enabled when the clock is advanced with a "C" keypress seems like a bug. Button components mapped to the keyboard also do not enable the stepping functions when their key is pressed. Both of these components only enable the functions when clicked.

_apologies for all of the edits_
